app/uk/gov/hmrc/gform/formtemplate/FormTemplateValidator.scala

```diff
@@ -155,8 +155,8 @@ object FormTemplateValidator {
     }
 
     formTemplate.authConfig match {
-      case HmrcSimpleModule | HmrcAgentModule(_) if !userContextComponentType.isEmpty =>
-        Invalid("Invalid auth type for component type.")
+      case HmrcSimpleModule | HmrcAgentModule(_) | Anonymous if userContextComponentType.nonEmpty =>
+        Invalid(s"You used ${formTemplate.authConfig} but you didn't provide 'serviceId'.")
```

Comment:
This validation should be reversed ie. we should match valid cases, not the invalid ones. Current implementation is wrong wrt to eeitt auth.
It should be something like this:
```
if (userContextComponentType.nonEmpty) {
  formTemplate.authConfig match {
    case HmrcEnrolmentModule(_) | HmrcAgentWithEnrolmentModule(_, _) => Valid
    case _ => Invalid("You used '${user.enrolledIdentifier}' but you didn't provide 'serviceId'.")
  }
} else Valid
```